Panic when starting Zeno with go run

Reported by @HarshNarayanJha in [85](https://github.com/internetarchive/Zeno/issues/85).

```
go run . get url https://some.url

panic: runtime error: slice bounds out of range [:7] with length 6

goroutine 1 [running]:
github.com/internetarchive/Zeno/internal/pkg/crawl.GenerateCrawlConfig(0xc000478008)
	/home/corentin/Documents/work/ia/Zeno/internal/pkg/crawl/config.go:248 +0x108b
github.com/internetarchive/Zeno/cmd.init.func8(0xc000418c00?, {0xc000428520, 0x1, 0x14da456?})
	/home/corentin/Documents/work/ia/Zeno/cmd/get_url.go:24 +0x39
github.com/spf13/cobra.(*Command).execute(0x21cdf40, {0xc0004284f0, 0x1, 0x1})
	/home/corentin/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0xaca
github.com/spf13/cobra.(*Command).ExecuteC(0x21cd6a0)
	/home/corentin/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/corentin/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/internetarchive/Zeno/cmd.Run()
	/home/corentin/Documents/work/ia/Zeno/cmd/cmd.go:56 +0x211
main.main()
	/home/corentin/Documents/work/ia/Zeno/main.go:21 +0x13
exit status 2

--- a/internal/pkg/crawl/config.go
+++ b/internal/pkg/crawl/config.go
@@ -245,6 +245,12 @@ func GenerateCrawlConfig(config *config.Config) (*Crawl, error) {
 		c.UserAgent = config.UserAgent
 	} else {
 		version := utils.GetVersion()
+
+		// If Version is a commit hash, we only take the first 7 characters
+		if len(version.Version) == 40 {
+			version.Version = version.Version[:7]
+		}
+
 		c.UserAgent = "Mozilla/5.0 (compatible; archive.org_bot +http://archive.org/details/archive.org_bot) Zeno/" + version.Version + " warc/" + version.WarcVersion
 	}
 	c.Headless = config.Headless

--- a/internal/pkg/crawl/config.go
+++ b/internal/pkg/crawl/config.go
@@ -245,12 +245,6 @@ func GenerateCrawlConfig(config *config.Config) (*Crawl, error) {
 		c.UserAgent = config.UserAgent
 	} else {
 		version := utils.GetVersion()
-
-		// If Version is a commit hash, we only take the first 7 characters
-		if len(version.Version) == 40 {
-			version.Version = version.Version[:7]
-		}
-
 		c.UserAgent = "Mozilla/5.0 (compatible; archive.org_bot +http://archive.org/details/archive.org_bot) Zeno/" + version.Version + " warc/" + version.WarcVersion
 	}
 	c.Headless = config.Headless

--- a/internal/pkg/crawl/config.go
+++ b/internal/pkg/crawl/config.go
@@ -245,7 +245,13 @@ func GenerateCrawlConfig(config *config.Config) (*Crawl, error) {
 		c.UserAgent = config.UserAgent
 	} else {
 		version := utils.GetVersion()
-		c.UserAgent = "Mozilla/5.0 (compatible; archive.org_bot +http://archive.org/details/archive.org_bot) Zeno/" + version.Version[:7] + " warc/" + version.WarcVersion
+
+		// If Version is a commit hash, we only take the first 7 characters
+		if len(version.Version) == 40 {
+			version.Version = version.Version[:7]
+		}
+
+		c.UserAgent = "Mozilla/5.0 (compatible; archive.org_bot +http://archive.org/details/archive.org_bot) Zeno/" + version.Version + " warc/" + version.WarcVersion
 	}
 	c.Headless = config.Headless
 

--- a/internal/pkg/utils/version.go
+++ b/internal/pkg/utils/version.go
@@ -13,7 +13,7 @@ type Version struct {
 
 func GetVersion() (version Version) {
 	// Defaults to "unknown"
-	version.Version = "unknown"
+	version.Version = "unknown_version"
 
 	if info, ok := debug.ReadBuildInfo(); ok {
 		// Determine Zeno's version based on Git data

--- a/internal/pkg/utils/version.go
+++ b/internal/pkg/utils/version.go
@@ -12,8 +12,8 @@ type Version struct {
 }
 
 func GetVersion() (version Version) {
-	// Defaults to master
-	version.Version = "master"
+	// Defaults to "unknown"
+	version.Version = "unknown"
 
 	if info, ok := debug.ReadBuildInfo(); ok {
 		// Determine Zeno's version based on Git data

--- a/internal/pkg/utils/version.go
+++ b/internal/pkg/utils/version.go
@@ -12,7 +12,7 @@ type Version struct {
 }
 
 func GetVersion() (version Version) {
-	// Defaults to "unknown"
+	// Defaults to "unknown_version"
 	version.Version = "unknown_version"
 
 	if info, ok := debug.ReadBuildInfo(); ok {

